lib/charms/layer/basic.py

```diff
@@ -4,11 +4,19 @@
 from glob import glob
 from subprocess import check_call
 
+from charms.layers.execd import execd_preinstall
+
 
 def bootstrap_charm_deps():
     """
     Set up the base charm dependencies so that the reactive system can run.
     """
+    # execd must happen first, before any attempt to install packages or
```

Comment:
I'm late to the party, but you just unlocked what exec.d / basenode really is used for. Thanks for the breakdown stub!